build.gradle

```diff
@@ -112,28 +113,86 @@ dependencies {
     testImplementation "org.hamcrest:hamcrest:${hamcrest}"
 }
 
-test {
-    testLogging {
-        showStandardStreams = true // disable stdout & stderr redirection
-    }
-    useJUnitPlatform()
+// install the CLI, either from source code directly or from a GH release, before starting tests
+task runInstallForTesting(type: Exec){
+    boolean testInstallFromGitHub = project.hasProperty("testInstallFromGitHub");
+    String installMode = testInstallFromGitHub ? "GITHUB_RELEASE" : "SOURCE_CODE"
+    commandLine "sh", "./tools/install-for-testing.sh", installMode
+
+    dependsOn installDist
 }
 
-task unitTest(type: Test) {
-    useJUnitPlatform {
-        includeTags 'unit'
+// run tests with the specified tag and install mode.
+// 1) run unit tests directly against the source code
+//    ./gradlew runTestsWithTag -PtestTag=unit
+// 2) run integration tests against an installation built from source code
+//    ./gradlew runTestsWithTag -PtestTag=integration
+// 3) run integration tests against an installation built from the latest GitHub release
+//    ./gradlew runTestsWithTag -PtestTag=integration -PtestInstallFromGitHub
+task runTestsWithTag(type: Test) {
+    beforeTest { descriptor ->
+        println "========================================================="
+        println "Running test: ${descriptor.name} [${descriptor.className}]"
+    }
+    afterTest { descriptor, result ->
+        println "Finished running test ${descriptor.name} [${descriptor.className}] with result: ${result.resultType}"
+    }
+    testLogging {
+        showStandardStreams = true // disable stdout & stderr redirection
     }
-    // force tests to always be re-run
-    outputs.upToDateWhen { false }
 
     doFirst {
-        if(!project.hasProperty("useDefaultContextDir")) {
+        // gradle project properties define which tests to run and how to do the install before running them
+        String testTag = project.findProperty("testTag");
+        boolean testInstallFromGitHub = project.hasProperty("testInstallFromGitHub");
+        boolean writeContextToHomeDir = project.hasProperty("writeContextToHomeDir")
+
+        // if the test tag is not defined, then throw an error. we don't know what tests to run
+        if (testTag == null) {
```

Comment:
I prefer to be explicit about what tests are being run, since there doesn't seem to be an obvious default. I think it makes the GH actions in particular easier to follow.